Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("rootfile1.txt");
-
-    for (const QString& expect : expectedFiles)
+    TEST_F(AssetScannerUnitTest, AssetScanner_ScanMultipleFolders_ExpectedFilesAndFoldersFound)
     {
-        UNIT_TEST_EXPECT_TRUE(CreateDummyFile(expect));
-    }
-
-    // but we're going to not watch subfolder3 recursively, so... remove these files (even though they're on disk)
-    // if this causes a failure it means that its ignoring the "do not recurse" flag:
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt"));
-
-    QSet<QString> expectedFolders;
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg");
-
-
-    PlatformConfiguration config;
-    AZStd::vector<AssetBuilderSDK::PlatformInfo> platforms;
-    config.PopulatePlatformsForScanFolder(platforms);
-    //                                               PATH               DisplayName  PortKey outputfolder  root recurse  platforms
-    config.AddScanFolder(ScanFolderInfo(tempPath.absolutePath(),         "temp",       "ap1",    true,  false, platforms));  // note:  "Recurse" set to false.
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder1"), "",           "ap2",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder2"), "",           "ap3",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder3"), "",           "ap4",   false,  false, platforms)); // note:  "Recurse" set to false.
-    AssetScanner scanner(&config);
-
-    QList<AssetFileInfo> filesFound;
-    QList<AssetFileInfo> foldersFound;
-
-    bool doneScan = false;
-
-    connect(&scanner, &AssetScanner::FilesFound, this, [&filesFound](QSet<AssetFileInfo> fileList)
+        using namespace AssetProcessor;
+        QSharedPointer<QCoreApplication> m_qApp;
+
+        // QApplications require command line arguments, so create something to meet that requirement.
+        int argC{ 1 };
+        char commandLineBuffer[AZ_MAX_PATH_LEN];
+        char* commandLineBufferAddress{ commandLineBuffer };
+        azstrcpy(commandLineBuffer, AZ_ARRAY_SIZE(commandLineBuffer), "no_argv_supplied");
+
+        m_qApp.reset(new QApplication(argC, &commandLineBufferAddress));
```

Comment:
`QCoreApplication` has [specific support for a null `argc`/`argv`](https://github.com/qt/qtbase/blob/v5.15.1/src/corelib/kernel/qcoreapplication.cpp#L449-L453). You can do this the same way [the EMotionFX tests do](https://github.com/o3de/o3de/blob/6948e637e0c715db7c6c2bd2ddb0ce57bf503176/Gems/EMotionFX/Code/Tests/UI/UIFixture.cpp#L50-L51).

```cpp
        int argc = 0;
        m_qApp = new QApplication(argc, nullptr);
```

Reply:
This documentation https://doc.qt.io/qt-5/qapplication.html#QApplication says that argc must be greater than zero, and argv must have a valid string

Reply:
Yeah, I've seen that in the documentation. The source says differently though, and it is a lot simpler passing `nullptr` for `argv` than it is managing a hand-rolled `argc`/`argv`. If you want to keep a non-null `argv`, I think this is clearer:

```cpp
int argc{ 1 };
char *argv[] = {"AssetScannerTest", nullptr};
m_qApp = new QApplication(argc, &argv[0]);
```

Note that `argv[argc]` is supposed to return `nullptr`.

Reply:
I've updated it to use 0 and nullptr, with a comment mentioning that it contradicts Qt's documentation.